cmd/kwild/root/root.go

```diff
@@ -112,6 +121,57 @@ func RootCmd() *cobra.Command {
 	return cmd
 }
 
+// parseExtensionFlags parses the extension flags from the command line and
+// returns a map of extension names to their configured values
+func parseExtensionFlags(args []string) (map[string]map[string]string, error) {
+	exts := make(map[string]map[string]string)
+	for i := 0; i < len(args); i++ {
+		if !strings.HasPrefix(args[i], "--") {
+			return nil, fmt.Errorf("expected extension flag, got %q", args[i])
+		}
+		// split the flag into the extension name and the flag name
+		// we intentionally do not use SplitN because we want to verify
+		// there are exactly 2 parts.
+		parts := strings.Split(args[i], ".")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid extension flag %q", args[i])
+		}
+
+		extName := strings.TrimPrefix(parts[0], "--")
+
+		// get the extension map for the extension name.
+		// if it doesn't exist, create it.
+		ext, ok := exts[extName]
+		if !ok {
+			ext = make(map[string]string)
+			exts[extName] = ext
+		}
+
+		// we now need to get the flag value. Flags can be passed
+		// as either "--extension.extname.flagname value" or
+		// "--extension.extname.flagname=value"
+		if strings.Contains(parts[1], "=") {
+			// flag value is in the same argument
+			val := strings.SplitN(parts[1], "=", 2)
```

Comment:
what about `--ext.name=`?  Since `--ext.name --ext.xx=2` will error, to make it consistent, I think here should check `val[1]` has value.

Reply:
This matches Cobra's parsing

Reply:
yeah you're right